hibernate-core/src/main/java/org/hibernate/sql/ast/tree/predicate/FilterPredicate.java

```diff
@@ -155,5 +159,36 @@ public JdbcMappingContainer getExpressionType() {
 		public boolean isEmpty() {
 			return false;
 		}
+
+		private Object retrieveParamValue(FilterImpl filter, String paramName, BeanContainer beanContainer) {
+			Class<? extends Supplier> clazz = filter.getParameterResolver( paramName );
+			if (clazz.isInterface()) {
+				return filter.getParameter( paramName );
```

Comment:
Regardless of whether there is a parameter resolver, an explicit filter value should IMO always take precedence.